src/Jackalope/Version/VersionHistory.php

```diff
@@ -214,9 +230,45 @@ public function hasVersionLabel($label, $version = null)
      */
     public function getVersionLabels($version = null)
     {
-        // @codeCoverageIgnoreStart
-        throw new NotImplementedException();
-        // @codeCoverageIgnoreEnd
+        if (is_null($this->versionLabels)) {
+            $this->versionLabels = array();
+            $node = $this->getNode('jcr:versionLabels');
+            foreach ($node->getProperties() as $property) {
+                /* @var Property $property */
+                if ($property->getName()!= "jcr:primaryType") {
+                    $name = $property->getName();
+                    $value = $this->objectManager->getNodeByIdentifier($property->getValue()->getIdentifier(), 'Version\\Version');
+                    $this->versionLabels[$name] = $value;
+                }
+            }
+        }
+
+        if ($version === null) {
+            return array_keys($this->versionLabels);
+        } else {
+            $versions = $this->getAllVersions();
+            $versionIsInHistory = false;
+
+            foreach ($versions as $versionCheck) {
+                /* @var VersionInterface $versionCheck */
+                if ($versionCheck->getIdentifier() == $version->getIdentifier()) {
```

Comment:
We altered the interface VersioningInterface for the transport client.
Maybe we can add the typehint here too?